src/elastio-snap.c

```diff
@@ -5395,53 +5559,70 @@ static inline unsigned long disable_page_protection(void) {
 static inline void reenable_page_protection(unsigned long cr0) {
 	wp_cr0(cr0);
 }
+#endif
 
-static void **find_sys_call_table(void){
-	long long offset;
-	void **sct;
-
-	if(!SYS_CALL_TABLE_ADDR || !SYS_MOUNT_ADDR || !SYS_UMOUNT_ADDR) return NULL;
-
-	offset = ((void *)kfree) - (void *)KFREE_ADDR;
-	sct = (void **)SYS_CALL_TABLE_ADDR + offset / sizeof(void **);
+/** generic function to manage system call table page permissions */
+static inline long syscall_mode_rw(void **syscall_table, int syscall_num)
+{
+#if defined(CONFIG_X86_64)
+	return disable_page_protection(); 
+#elif defined(CONFIG_ARM64)
+	return set_page_rw((unsigned long) (syscall_table + syscall_num));
+#else
+	return -1;
```

Comment:
I dislike this `-1` usage as an error sign for arm64 in the place of memory protection state for x86_64.
Please see other related comments.

Reply:
Accepted.